docs/frontend_architecture/testing_layout.rst

```diff
@@ -0,0 +1,186 @@
+Testing template
+================
+
+This is a common template that should serve as a nice starter for any test file testing a ``Vue.js`` component, and also gives a brief description on mocking the various dependencies of a common.
+
+The general template of a test file will be like:
+
+.. code:: javascript
+
+   import { render, screen } from '@testing-library/vue';
+   import userEvent from '@testing-library/user-event'
+   import YourExampleComponent from '../YourExampleComponent.vue';
+
+   const samplePropA = "someTestingValue1"
+   const samplePropB = "someTestingValue2"
+
+   // Helper function to render the component with the provided props
+   const renderComponent = props => {
+     return render(YourExampleComponent, {
+       props: {
+           propA: samplePropA,
+           propB: samplePropB,
+           ...props
+       }
+     });
+   };
+
+   describe('YourExampleComponent', () => {
+     test("smoke test", () => {
+       renderComponent();
+
+       // Make a basic assertion about the outermost container in the smoke test
+       expect(screen.getByText("Sample Component")).toBeInTheDocument();
+     })
+
+     test('describe some use case', async () => {
+       renderComponent({
+           propA: samplePropA
+       });
+
+       // Perform some user interaction
+       await userEvent.click(screen.getByRole("button", { name: "Click" }))
+
+       expect(screen.getByText(samplePropA)).toBeInTheDocument();
+     });
+   });
+
+Some quick notes about the code snippet:
+
+1. You would need to replace the ``YourExampleComponent`` with the actual component you want to test.
+2. It is usually a good practice to have the first test as a smoke test in all your components.
```

Comment:
I think this list is actually more Style Guides, so I think is better to have these items as subItems of the Style Guides section:
1. Add smoke tests: Add a smoke test to every test suites that only renders the most basic example of a component, where the only thing about assert is that the simplest render does not throw an error.
2. Use describe blocks: Use describe blocks to group unit tests that test similar functionality.
3. Avoid long and complex unit tests: A unit test should be kept simple and test a single execution flow. So that it is easy for someone else to read the test and understand the functionality of the component.
4. Use default props that are not relevant to your unit test: Within your `renderComponents` function declare default props so that you do not have to define them in unit tests where they are not relevant. (Here we could also give an example about this, although it is less important than the others).
5. Explicitly declare props that are relevant to your unit test: Avoid using default props for inputs that are relevant to your unit tests, and instead declare them explicitly in your test. For example, if we are testing the rendering of a list, then let's explicitly declare the list in the unit test. Let's not wait until someone else reading the test has to go to the `renderComponent` function to see what the unit test input was.

The wording could change, but I've more or less written down the ideas that could go in these subsections. With this I would think that this list of "quick notes about the code snippet" could be eliminated. Let me know what you think.

Reply:
Yes, it makes complete sense. Reading these points explicitly under the style guide would have better effect in terms of communication of these ideas. Thanks for taking out the time to articulate all these points! Will work on updating them in the documentation.